webcommon/typescript.editor/bundles/prepare/src/main/java/org/netbeans/modules/learning/prepare/PrepareBundles.java

```diff
@@ -163,27 +166,32 @@ public static void main(String... args) throws IOException, InterruptedException
                     });
                 }
                 MessageDigest md = MessageDigest.getInstance("SHA1");
-                try (InputStream in = Files.newInputStream(bundle)) {
-                    md.update(in.readAllBytes());
-                }
+                md.update(Files.readAllBytes(bundle));
                 StringBuilder hash = new StringBuilder();
                 for (byte b : md.digest()) {
                     hash.append(String.format("%02X", b));
                 }
                 Path external = externalDir.resolve(hash + "-" + bundle.getFileName());
                 Files.copy(bundle, external);
-                binariesList.write(hash + " " + bundle.getFileName().toString() + nl);
+                binaries.put(bundle.getFileName().toString(), hash.toString());
             }
         }
-        
+
+        try(Writer binariesList = new OutputStreamWriter(Files.newOutputStream(bundlesDir.resolve("binaries-list")), "UTF-8")) {
+            binariesList.write(readResourceIntoString("ALv2Header.txt"));
+            for(Entry<String,String> e: binaries.entrySet()) {
+                binariesList.write(e.getValue() + " " + e.getKey() + nl);
+            }
+        }
+
         Map<String, String> project2LicenseKey = new HashMap<>();
```

Comment:
Consider converting to `TreeMap`, too as the `-license.txt` is created in entrySet-iteration order.

Reply:
These files are independent from each other, so creation order of the license files does not matter for minimizing the diff. The order only matters in the binaries-list. But maybe I'm overlooking something?

Reply:
Nothing overly important, just a predictable ordering of the output.